src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/inmemory/PetRepositoryInMemoryInterpreter.scala

```diff
@@ -44,7 +44,7 @@ class PetRepositoryInMemoryInterpreter[F[_]: Applicative] extends PetRepositoryA
     cache.values.filter(p => statuses.exists(_ == p.status)).toList.pure[F]
 
   def findByTag(tags: NonEmptyList[String]): F[List[Pet]] =
-    cache.values.filter(p => tags.exists(_ == p.tags)).toList.pure[F]
+    cache.values.filter(p => p.tags.exists(pt => tags.exists(_ == pt))).toList.pure[F]
```

Comment:
yea, that definitely looks like a bug.  The nested `exists` is concerning here.

A better solution maybe to have a `tags` index, where each tag in the index points to a `NonEmptyList[String]`?

Reply:
What should be found is the intersection of the tags within a `Pet` (`Set[String]`) and the filter predicate (`NonEmptyList[String]`), right? Actually the method name `findByTag` sounds like initially the parameter was intended to be just one tag (a `String`). Maybe the logic could be simplified in that way?

Reply:
I think this can be slightly simplified by converting the param to a `Set` one time before the filter (as opposed to the `toList` in the filter)...

```scala
    val tagSet = tags.toList.toSet
    cache.values.filter { p => p.tags.intersect(tagSet).nonEmpty }.toList.pure[F]
```

Reply:
Avoiding computation of the set intersection would be a tiiiiny bit faster?
```
val tagSet = tags.toNes  // NonEmptyList => NonEmptySet
cache.values.filter(_.tags.exists(tagSet contains_ _)).toList.pure[F]
```

Reply:
oh man, too many underscores, can we at least add parens there :)